src/nyx/features/image_matrix.cpp

```diff
@@ -120,6 +102,68 @@ void ImageMatrix::print (const std::string& head, const std::string& tail, std::
 	std::cout << tail;
 }
 
+void ImageMatrix::print (std::ofstream& f, const std::string& head, const std::string& tail, std::vector<PrintablePoint> special_points)

```

Comment:
Should we pass the vector as a reference?

Reply:
Here - no, otherwise it'd ruin the initializer list based- default parameter logic:
void print(const std::string& head = "", const std::string& tail = "", std::vector<PrintablePoint> special_points = {});
(See image_matrix.h)